vulcanai2/models/basenetwork.py

```diff
@@ -61,14 +61,18 @@ def __init__(self, name, dimensions, config, save_path=None, input_networks=None
         super(BaseNetwork, self).__init__()
 
         self._name = name
-        if isinstance(dimensions, tuple):
-            self._dimensions = [dimensions]
-        elif isinstance(dimensions, int):
-            self._dimensions = (dimensions,)
+        if dimensions is not None:
+            # Must be a list of tuples(ConvNet)/int(DenseNet)
+            if isinstance(dimensions, (tuple, int)):
+                self.in_dim = [dimensions] 
```

Comment:
why wrap it in a list if it is already a tuple?

Reply:
didn't we agree on list of tuples?